core/src/test/scala/org/apache/spark/scheduler/DAGSchedulerSuite.scala

```diff
@@ -547,6 +554,136 @@ class DAGSchedulerSuite
     assert(sparkListener.failedStages.size == 1)
   }
 
+  /** This tests the case where another FetchFailed comes in while the map stage is getting
```

Comment:
nit: we generally use javadoc-style (even though this is not a method declaration):

    /**
     * Comment.
     */